client/webserver/site/src/js/forms.js

```diff
@@ -298,31 +302,91 @@ export class WalletConfigForm {
  * ConfirmRegistrationForm should be used with the "confirmRegistrationForm" template.
  */
 export class ConfirmRegistrationForm {
-  constructor (application, form, { getDexAddr, getCertFile }, success, insufficientFundsFail) {
+  constructor (application, form, { getDexAddr, getCertFile }, success, insufficientFundsFail, setupWalletFn) {
     this.fields = Doc.parsePage(form, [
-      'feeDisplay', 'marketRowTemplate', 'marketsTableRows', 'appPass', 'appPassBox',
-      'appPassSpan', 'submitConfirm', 'regErr'
+      'marketRowTemplate', 'marketsTableRows', 'appPass', 'appPassBox',
+      'appPassSpan', 'submitConfirm', 'regErr', 'feeRowTemplate',
+      'feeTableRows', 'setupWallet'
     ])
     app = application
     this.getDexAddr = getDexAddr
     this.getCertFile = getCertFile
     this.success = success
     this.insufficientFundsFail = insufficientFundsFail
     this.form = form
-    bind(form, this.fields.submitConfirm, () => this.submitForm())
+    this.setupWalletFn = setupWalletFn
+    this.feeWallet = null
+    bind(form, this.fields.submitConfirm, () => this.submitForm(this.feeWallet))
+  }
+
+  setWalletFee (assetID) {
+    this.feeWallet = assetID
   }
 
   /*
    * setExchange populates the form with the details of an exchange.
    */
   setExchange (xc) {
+    // store xc in case we need to refresh the data, like after setting up
+    // a new wallet.
+    this.xc = xc
     const fields = this.fields
     this.fees = xc.regFees
-    const dcr = this.fees.dcr // TODO: display all options and give choice
-    fields.feeDisplay.textContent = `${Doc.formatCoinValue(dcr.amount / 1e8)} ${app.assets[dcr.id].symbol.toUpperCase()}`
     while (fields.marketsTableRows.firstChild) {
       fields.marketsTableRows.removeChild(fields.marketsTableRows.firstChild)
     }
+    while (fields.feeTableRows.firstChild) {
+      fields.feeTableRows.removeChild(fields.feeTableRows.firstChild)
+    }
+    for (const [symbol, fee] of Object.entries(xc.regFees)) {
+      // if asset fee is not supported by the client we can skip it.
+      if (app.user.assets[fee.id] === undefined) {
+        continue
+      }
+      const haveWallet = app.user.assets[fee.id].wallet
+      const tr = fields.feeRowTemplate.cloneNode(true)
+      Doc.bind(tr, 'click', () => {
+        this.setWalletFee(fee.id)
```

Comment:
See other comment about `setFeeWallet`, but here you can just do `this.feeWallet = assetID`, or `this.feeAssetID = assetID` if you go with my naming suggestion.